### PR TITLE
Meta image logic fix

### PR DESCRIPTION
**Pull request by mercuree**
1. Fixed condition:
```python
if try_one is None:
```
which is always False, because 
```python
 try_one = self.get_meta_content(doc, 'meta[property="og:image"]')
```
never returns None

2. Changed the condition order. Icon image is least relevant for top_img

### CODE REVIEW

File: newspaper/extractors.py [hunk 1]
```diff
@@ -429,19 +429,19 @@ def get_meta_img_url(self, article_url, doc):
         """
         top_meta_image, try_one, try_two, try_three, try_four = [None] * 5
         try_one = self.get_meta_content(doc, 'meta[property="og:image"]')
-        if try_one is None:
-            link_icon_kwargs = {'tag': 'link', 'attr': 'rel', 'value': 'icon'}
-            elems = self.parser.getElementsByTag(doc, **link_icon_kwargs)
-            try_two = elems[0].get('href') if elems else None
-
-        if try_two is None:
+        if try_one == '':
```
Comment (yprez) [inline newspaper/extractors.py]: `if not try_one` would be more elegant imo

Comment (yprez) [inline newspaper/extractors.py]: Would also make the next conditions simpler.

Comment (mercuree) [inline newspaper/extractors.py]: You are right. When you replace `if try_one is none` with `if not try_one`, the difference is unclear, and i wrote more explicit way.

**Comment by codelucas** [comment]
This is a fantastic fix @mercuree. Also great work on the unit tests 👍 💯 Double checking and merging it now!

**Comment by codelucas** [comment]
Wait, this appears to be failing the CI build - the failure is in the new unit test you added: 
https://travis-ci.org/codelucas/newspaper/jobs/279784081

Weird how it passed the CI build locally in the comment though

**Comment by codelucas** [comment]
I'll take a look

**Comment by codelucas** [comment]
Haha problem found! Ironically the new bug was due to an optimization from your earlier PR!

Your PR here (https://github.com/codelucas/newspaper/pull/355) improved our xml parse optimization by turning regex off .. BUT this PR was based on the code before your optimization and thus does not turn on the regex api.

new code in this PR:
```
 elems = self.parser.getElementsByTag(doc, **link_img_src_kwargs)
```

I'll submit a PR later tonight before deploying a new version that re-enables regex operations (or performs two queries)

**Pull request merged**